freqtrade/data/converter/orderflow.py

```diff
@@ -256,34 +256,24 @@ def trades_orderflow_to_imbalances(df: pd.DataFrame, imbalance_ratio: int, imbal
     return dataframe
 
 
-def stacked_imbalance(
-    df: pd.DataFrame, label: str, stacked_imbalance_range: int, should_reverse: bool
-):
+def stacked_imbalance(df: pd.DataFrame, label: str, stacked_imbalance_range: int):
     """
     y * (y.groupby((y != y.shift()).cumsum()).cumcount() + 1)
     https://stackoverflow.com/questions/27626542/counting-consecutive-positive-values-in-python-pandas-array
     """
     imbalance = df[f"{label}_imbalance"]
     int_series = pd.Series(np.where(imbalance, 1, 0))
-    stacked = int_series * (
-        int_series.groupby((int_series != int_series.shift()).cumsum()).cumcount() + 1
-    )
-
-    max_stacked_imbalance_idx = stacked.index[stacked >= stacked_imbalance_range]
-    stacked_imbalance_price = np.nan
-    if not max_stacked_imbalance_idx.empty:
-        idx = (
-            max_stacked_imbalance_idx[0]
-            if not should_reverse
-            else np.flipud(max_stacked_imbalance_idx)[0]
-        )
-        stacked_imbalance_price = imbalance.index[idx]
-    return stacked_imbalance_price
-
-
-def stacked_imbalance_ask(df: pd.DataFrame, stacked_imbalance_range: int):
-    return stacked_imbalance(df, "ask", stacked_imbalance_range, should_reverse=True)
-
-
-def stacked_imbalance_bid(df: pd.DataFrame, stacked_imbalance_range: int):
-    return stacked_imbalance(df, "bid", stacked_imbalance_range, should_reverse=False)
+    # Group consecutive True values and get their counts
+    groups = (int_series != int_series.shift()).cumsum()
+    counts = int_series.groupby(groups).cumsum()
+
+    # Find indices where count meets or exceeds the range requirement
+    valid_indices = counts[counts >= stacked_imbalance_range].index
+
+    stacked_imbalance_prices = []
+    if not valid_indices.empty:
+        # Get all prices from valid indices from beginning of the range
+        stacked_imbalance_prices = [
+            imbalance.index.values[idx - (stacked_imbalance_range - 1)] for idx in valid_indices
+        ]
+    return stacked_imbalance_prices if stacked_imbalance_prices else []
```

Comment:
you shouldn't need this - stacked_imbalance_prices is already `[]` if the grouping yields empty.

This change might also warant a small update to the docs? 
![image](https://github.com/user-attachments/assets/ffed4049-8f1d-4c3b-8e19-5a73b921451c)

sounds slightly wrong now  - as we're now returning an array.